main.rb

```diff
@@ -223,27 +228,37 @@ def move_train_backward
     puts '-----'
   end
 
-
   def show_stations
     self.stations.each_with_index { |station, index| puts "[#{index}] #{station.name}" }
     puts '-----'
   end
 
-  def show_trains_on_station
-    return if self.trains.empty? || self.stations.empty?
 
-    self.show_stations
-    station_index = get_station_index
-    
-    return if station_index.nil? || self.stations[station_index].nil?
+  def show_trains_on_one_station
+    return if stations.empty?
+    station = get_station
+    show_trains_on_station(station)
+  end
 
-    trains = [] 
-    self.stations[station_index].trains.each { |train| trains << train }
-    show_trains(trains)
+  def show_trains_on_each_station
+    return if stations.empty?
+
+    stations.each do |station|
+      show_trains_on_station(station)
+      puts
+    end
 
-    puts '-----'
   end
-      
+ 
+ def show_trains_on_station(station)
+    puts "Станция: #{station.name} (поездов: #{station.trains.length})"
+    station.each_train do |train|
+      puts train.to_s
```

Comment:
`puts` автоматически вызывает у объекта метод `to_s`
можно так:
```suggestion
      puts train
```
вот: 
```
$ irb
2.6.3 :001 > class A; def to_s; 'KUKU'; end; end;
 => :to_s 
2.6.3 :002 > a = A.new
 => #<A:0x000055b50dcd4390> 
2.6.3 :003 > puts a
KUKU
```